Grant Application - Reference blockchain system.md

```diff
@@ -0,0 +1,286 @@
+
+## Development Roadmap :nut_and_bolt:
+
+**Expected deliverables:**
+
+1. A reference token standard that records and reflects the nominal reference relationship between new work and its referred old work. When creating a new work, the corresponding new token will also be generated and attached where the reference relationship between the new work and its referred work will be added and stored. 
+
+2. A reference system that makes old work truly composable and interoperable with the new work. By referring to the old work, new work could be created by adding anything new or changing anywhere on the referred work while still connecting and sharing other parts of the referred work seamlessly. 
+
+3. A revenue allocation and distribution system that confirms the revenue distribution scheme agreed by different works with the reference relationship. It will track the earnings of the new work, and distribute them with the old work which the new work refers to according to their distribution scheme.
+
+All programs will be developed in !ink on Pallets.
+
+
+**Milestones:**
+
+### Overview
+
+- **Total Estimated Duration:** 6 months
+- **Full-Time Equivalent (FTE):**  2 ~ 4 people depending on actual workload
+- **Total Costs:** $20,000
+
+
+### Milestone 1 — Reference token standard
+
+- **Estimated duration:** 1 month
+- **FTE:**  2 developers and 1 technical researcher
+- **Costs:** 3,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | We will design a new reference license in our project, which records and reflects the reference relationship between the referred old works and the new works that are based on the old works. |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains the design and usage of our work in details. |
+|  1. | Specification | We will specify the details of our token standard design that how users can create their own work, refer to others' work, and get the earnings of their work via the reference token in the reference system. |
+|  2. | Setup | We provide a repository that forms the basis of our deliverables. |
+|  3. | Core Types | We provide the core data and coding types in details. |
```

Comment:
Can you be more specific here? Which types? Is this a full implementation or just a specification?

Reply:
Yes we will fully implement it in our demo. Core types mean the data types in the objects of our token standard. Will revise it more clearly.